docs/orchestrating-elastic-stack-applications/agent.asciidoc

```diff
@@ -0,0 +1,403 @@
+
+[id="{p}-elastic-agent-custom-configuration"]
+=== Customize Elastic Agent configuration
+
+The Elastic Agent configuration is defined in the `config` element:
+
+[source,yaml,subs="attributes,+macros"]
+----
+apiVersion: agent.k8s.elastic.co/v1alpha1
+kind: Agent
+metadata:
+  name: quickstart
+spec:
+  version: {version}
+  elasticsearchRefs:
+  - name: quickstart
+  daemonSet: {}
+  config:
+    inputs:
+      - name: system-1
+        revision: 1
+        type: system/metrics
+        use_output: default
+        meta:
+          package:
+            name: system
+            version: 0.9.1
+        data_stream:
+          namespace: default
+        streams:
+          - id: system/metrics-system.cpu
+            data_stream:
+              dataset: system.cpu
+              type: metrics
+            metricsets:
+              - cpu
+            cpu.metrics:
+              - percentages
+              - normalized_percentages
+            period: 10s
+----
+
+Alternatively, it can be provided via a Secret specified in the `configRef` element. The Secret must have an `agent.yml` entry with the configuration:
+[source,yaml,subs="attributes,+macros"]
+----
+apiVersion: agent.k8s.elastic.co/v1alpha1
+kind: Agent
+metadata:
+  name: quickstart
+spec:
+  version: {version}
+  elasticsearchRefs:
+  - name: quickstart
+  daemonSet: {}
+  configRef:
+    secretName: system-cpu-config
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: system-cpu-config
+stringData:
+  agent.yml: |-
+    inputs:
+      - name: system-1
+        revision: 1
+        type: system/metrics
+        use_output: default
+        meta:
+          package:
+            name: system
+            version: 0.9.1
+        data_stream:
+          namespace: default
+        streams:
+          - id: system/metrics-system.cpu
+            data_stream:
+              dataset: system.cpu
+              type: metrics
+            metricsets:
+              - cpu
+            cpu.metrics:
+              - percentages
+              - normalized_percentages
+            period: 10s
+----
+
+For more details especially on how to generate configuration for Elastic Agent in standalone mode, see the link:https://www.elastic.co/guide/en/fleet/current/run-elastic-agent-standalone.html[Elastic Agent standalone] documentation.
+
+
+[id="{p}-elastic-agent-multi-output"]
+=== Use multiple Elastic Agent outputs
+
+Elastic Agent supports the use of multiple outputs. The `elasticsearchRefs` element accepts therefore multiple references to Elasticsearch clusters. ECK populates the outputs section of the Elastic Agent configuration based on those references. If you configure more than one output you also have to specify a unique `outputName` attribute.
+
+To send Elastic Agent's internal monitoring and log data to a different Elasticsearch cluster called `agent-monitoring` and the harvested metrics to our `quickstart` cluster define two `elasticsearchRefs` as follows (abbreviated example):
```

Comment:
Maybe mention that the example assumes the same namespace? Or provide the namespace (could be different for the purposes of the example) in the manifest?